src/goto-instrument/accelerate/scratch_program.cpp

```diff
@@ -71,7 +71,12 @@ bool scratch_programt::check_sat(bool do_slice, guard_managert &guard_manager)
   std::cout << "Finished symex, invoking decision procedure.\n";
 #endif
 
-  return ((*checker)() == decision_proceduret::resultt::D_SATISFIABLE);
+  auto result = (*checker)();
+  INVARIANT(
```

Comment:
`INVARIANT` seems wrong here -- errors do happen, and should be dealt with by giving a proper error message to the user.

Think of `INVARIANT` as something that you could prove to be true, always.

Reply:
Thanks!

I am not super familiar with the code base, so I am not sure what other construct I could use here to report an error to the user. The function returns a `bool`, so the return value can't capture 3 possibilities: SAT, UNSAT, ERROR.

@kroening  @thomasspriggs do you have any suggestions?

Reply:
Looking at the call sites for this function, I realized that there is exception handling code around it, so probably the right thing to do here is to `throw` an exception and let the caller handle it properly.

Reply:
Added a `switch` as per Daniel's suggestion.